pvlib/iotools/srml.py

```diff
@@ -0,0 +1,183 @@
+
+    Notes
+    -----
+    Note that the time index is shifted back one minute to account for
+    2400 hours, and to avoid time parsing errors on leap years. Data values
+    on a given line should now be understood to occur during the interval
+    extending from the time of the line in which they are listed to
+    the ending time on the next line, rather than the previous line.
+
+    See SRML's `Archival Files`_ page for more information.
+
+    .. _Archival Files: http://solardat.uoregon.edu/ArchivalFiles.html
+
+    References
+    ----------
+
+    [1] University of Oregon Solar Radiation Monitoring Laboratory
+        `http://solardat.uoregon.edu/ <http://solardat.uoregon.edu/>`_
+    """
+    tsv_data = pd.read_csv(filename, delimiter='\t')
+    year = tsv_data.columns[1]
+    data = format_index(tsv_data, year)
+    # Drop day of year and time columns
+    data = data[data.columns[2:]]
+
+    data = data.rename(columns=map_columns)
+
+    # Quality flag columns are all labeled 0 in the original data. They
+    # appear immediately after their associated variable and are suffixed
+    # with an integer value when read from the file. So we map flags to
+    # the preceding variable with a '_flag' suffix.
+    #
+    # Example:
+    #   Columns ['ghi_0', '0.1', 'temp_air_2', '0.2']
+    #
+    #   Yields a flag_label_map of:
+    #       { '0.1': 'ghi_0_flag',
+    #         '0.2': 'temp_air_2'}
+    #
+    columns = data.columns
+    flag_label_map = {flag: columns[columns.get_loc(flag) - 1] + '_flag'
+                      for flag in columns[1::2]}
+    data = data.rename(columns=flag_label_map)
+
+    # Mask data marked with quality flag 99 (bad or missing data)
+    for col in columns[::2]:
+        missing = data[col + '_flag'] == 99
+        data[col] = data[col].where(~(missing), np.NaN)
+    return data
+
+
+def map_columns(col):
+    """Map data element numbers to pvlib names.
+
+    Parameters
+    ----------
+    col: str
+        Column label to be mapped.
+
+    Returns
+    -------
+    str
+        The pvlib label if it was found in the mapping,
+        else the original label.
+
+    """
+    if col.startswith('7'):
+        # spectral data
+        try:
+            return variable_map[col]
+        except KeyError:
+            return col
+    try:
+        variable_name = variable_map[col[:3]]
+        variable_number = col[3:]
+        return variable_name + '_' + variable_number
+    except KeyError:
+        return col
+
+
+def format_index(df, year):
+    """ Create a datetime index from day of year, and time columns.
+
+    Parameters
+    ----------
+    df: pd.Dataframe
+        The srml data to reindex.
+    year: int
+        The year of the file
+
+    Returns
+    -------
+    df: pd.Dataframe
+        The Dataframe with a datetime index applied.
+    """
+    df_time = df[df.columns[1]] - 1
+    df_doy = df[df.columns[0]]
+    fifty_nines = df_time % 100 == 99
```

Comment:
These variable names might be more confusing than I had intended. 59 is the desired result here, after reducing the times by 1 all of the hours become HH99(e.g. 2400->2399), so they need to be adjusted. 
This was to get around exceptions trying to parse a datetime from day 366 hour 2400 of a leap year. It is admittedly quite confusing. I will document the rationale better here.

Reply:
Aha, makes sense now.  Because of -1, the times run from hh00, hh01, ..., hh58, hh99.  Minor nit - putting `df_doy = ` ahead of  `df_time = ` would help me see the connection.